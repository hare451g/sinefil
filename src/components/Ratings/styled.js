import styled from 'styled-components';

const RatingContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

const Heading = styled.div`
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  line-height: 28px;
  padding-bottom: 12px;
`;

export { Heading, RatingContainer };
