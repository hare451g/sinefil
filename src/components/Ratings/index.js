import React from 'react';
import { Heading, RatingContainer } from './styled';

function Ratings({ ratings = [] }) {
  return (
    <>
      <Heading>Ratings</Heading>
      {ratings.length >= 1 ? (
        ratings.map(({ value, label }, index) => (
          <RatingContainer
            key={`rating-item-${index}`}
            data-testid={`rating-item-${index}`}
          >
            <div>{label}</div>
            <div>{value}</div>
          </RatingContainer>
        ))
      ) : (
        <RatingContainer data-testid="rating-empty-label">
          Movie is not yet rated
        </RatingContainer>
      )}
    </>
  );
}

export default Ratings;
