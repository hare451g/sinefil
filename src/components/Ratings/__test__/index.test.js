import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import Ratings from '../index';

describe('Movie Card Component', () => {
  const mockRatings = [
    {
      label: 'Internet Movie Database',
      value: '8.6 / 10',
    },
    {
      label: 'Internet Movie Database',
      value: '81%',
    },
    {
      label: 'Metacritic',
      value: '65 / 100',
    },
  ];

  it(
    'Should render without error' +
      ' and render Movie is not yet rated if there is no ratings provided',
    () => {
      const { getByTestId } = render(<Ratings />);
      const contents = getByTestId('rating-empty-label');
      expect(contents.innerHTML).toEqual('Movie is not yet rated');
    },
  );

  it('should render provided ratings keypair', () => {
    const { getAllByTestId } = render(<Ratings ratings={mockRatings} />);
    const contents = getAllByTestId(/rating-item-/i);

    expect(contents.length).toEqual(mockRatings.length);
    contents.forEach((content, index) => {
      expect(content.children).toMatchSnapshot(`
        <div>${mockRatings[index].label}</div>
        <div>${mockRatings[index].value}</div>
    `);
    });
  });
});
