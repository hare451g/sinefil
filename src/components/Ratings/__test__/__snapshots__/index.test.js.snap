// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card Component should render provided ratings keypair: 
        <div>Internet Movie Database</div>
        <div>8.6 / 10</div>
     1`] = `
HTMLCollection [
  <div>
    Internet Movie Database
  </div>,
  <div>
    8.6 / 10
  </div>,
]
`;

exports[`Movie Card Component should render provided ratings keypair: 
        <div>Internet Movie Database</div>
        <div>81%</div>
     1`] = `
HTMLCollection [
  <div>
    Internet Movie Database
  </div>,
  <div>
    81%
  </div>,
]
`;

exports[`Movie Card Component should render provided ratings keypair: 
        <div>Metacritic</div>
        <div>65 / 100</div>
     1`] = `
HTMLCollection [
  <div>
    Metacritic
  </div>,
  <div>
    65 / 100
  </div>,
]
`;
